tests/utils.go

```diff
@@ -242,11 +289,11 @@ func createAndVerifyPVC() {
 		OpenEBSNamespace,
 	)
 
-	ginkgo.By("verifying pvc status as bound")
+	ginkgo.By("verifying pvc status as bound\n")
+	ok := IsPVCBoundEventually(pvcName, 15, 3)
 
-	status := IsPVCBoundEventually(pvcName)
-	gomega.Expect(status).To(gomega.Equal(true),
-		"while checking status equal to bound")
+	gomega.Expect(ok).To(gomega.Equal(expect_bound),
```

Comment:
Seems the not bound here is simple relying on it not being bound within a timeout?
Could we check for expected errors instead?

Reply:
k8s doesnt append any errors in PVC. We can check only Phase (Bound or Pending) to determine the provisioning. We wait for 15 seconds with 3 secs poll. Ive noticed it does 3 to 4 retries within first 15 secs.

Reply:
I might be misremembering but isn't there an error message on the pvc events or something like that?

Reply:
We get these events on pvc:

```
Events:
  Type     Reason                Age                From                                                                                               Message
  ----     ------                ----               ----                                                                                               -------
  Normal   ExternalProvisioning  11s (x4 over 24s)  persistentvolume-controller                                                                        Waiting for a volume to be created either by the external provisioner 'local.csi.openebs.io' or manually by the system administrator. If volume creation is delayed, please verify that the provisioner is running and correctly registered.
  Warning  ProvisioningFailed    4s (x5 over 21s)   local.csi.openebs.io_openebs-lvm-controller-546768b57c-drszh_3c77800c-663f-469c-a3e5-be02414c02b8  failed to provision volume with StorageClass "openebs-lvmpv-non": rpc error: code = ResourceExhausted desc = no vg available to serve volume request having regex="^foo1n$" & capacity="4294967296"
  Normal   Provisioning          1s (x6 over 24s)   local.csi.openebs.io_openebs-lvm-controller-546768b57c-drszh_3c77800c-663f-469c-a3e5-be02414c02b8  External provisioner is provisioning volume for claim "default/csi-lvmpv"
```

However, im thinking just checking PVC state to be not Bound should suffice. We also validate lvmvolume cr for negative scenario after this.